backend/Application/Api/GraphQL/Search/SearchResult.cs

```diff
@@ -98,4 +115,70 @@ public IEnumerable<NodeStatus> GetNodeStatuses([Service] NodeStatusSnapshot node
         return nodeSummarySnapshot.NodeStatuses
             .Where(x => x.NodeName != null && x.NodeName.Contains(_queryString, StringComparison.InvariantCultureIgnoreCase));
     }
+    
+    /// <summary>
+    /// Try match query with contract regex.
+    /// 
+    /// Only consider query if there is a match.
+    /// </summary>
+    internal static bool TryMatchContractPattern(
+        string? query, out string? pattern)
+    {
+        const char end = '%';
+        const char start = '<';
+        const char comma = ',';
+        
+        pattern = null;
+        if (query is null)
+        {
+            return false;
+        }
+        
+        var match = ContractAddressRegex.Match(query);
+        if (!match.Success) return false;
+
+        switch (match.Groups[2].Success)
+        {
+            // The query ends in a comma or comma + space
+            case true when string.IsNullOrEmpty(match.Groups[2].Value):
+            {
+                var firstSpan = match.Groups[1].ValueSpan;
```

Comment:
You have to read through each scope of the switch case, to understand what's happening.
Extract them to a method, explaining what is going on in the method name.
When extracting them to a method it will also allow you to - reuse the code in the first two cases, as they do not differ a lot